pkg/beholder/client.go

```diff
@@ -0,0 +1,306 @@
+package beholder
+
+import (
+	"context"
+	"errors"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/exporters/otlp/otlplog/otlploggrpc"
+	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetricgrpc"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	otellog "go.opentelemetry.io/otel/log"
+	otelmetric "go.opentelemetry.io/otel/metric"
+	sdklog "go.opentelemetry.io/otel/sdk/log"
+	sdkmetric "go.opentelemetry.io/otel/sdk/metric"
+	sdkresource "go.opentelemetry.io/otel/sdk/resource"
+	"go.opentelemetry.io/otel/sdk/trace"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	oteltrace "go.opentelemetry.io/otel/trace"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+type Emitter interface {
+	// Sends message with bytes and attributes to OTel Collector
+	Emit(ctx context.Context, body []byte, attrKVs ...any) error
+}
+type Client interface {
+	Close() error
+}
+
+type messageEmitter struct {
+	messageLogger otellog.Logger
+}
+
+type OtelClient struct {
+	Config Config
+	// Logger
+	Logger otellog.Logger
+	// Tracer
+	Tracer oteltrace.Tracer
+	// Meter
+	Meter otelmetric.Meter
+	// Message Emitter
+	Emitter Emitter
+
+	// Providers
+	LoggerProvider        otellog.LoggerProvider
+	TracerProvider        oteltrace.TracerProvider
+	MeterProvider         otelmetric.MeterProvider
+	MessageLoggerProvider otellog.LoggerProvider
+
+	// OnClose
+	OnClose func() error
+}
+
+// NewOtelClient creates a new Client with OTel exporter
+func NewOtelClient(cfg Config, errorHandler errorHandlerFunc) (OtelClient, error) {
+	factory := func(ctx context.Context, options ...otlploggrpc.Option) (sdklog.Exporter, error) {
+		return otlploggrpc.New(ctx, options...)
+	}
+	return newOtelClient(cfg, errorHandler, factory)
+}
+
+// Used for testing to override the default exporter
+type otlploggrpcFactory func(ctx context.Context, options ...otlploggrpc.Option) (sdklog.Exporter, error)
+
+func newOtelClient(cfg Config, errorHandler errorHandlerFunc, otlploggrpcNew otlploggrpcFactory) (OtelClient, error) {
+	ctx := context.Background()
```

Comment:
It's unfortunately that a context is required for construction, but let's include it as an argument so we can still short circuit start-up if necessary:
```suggestion
func newOtelClient(ctx context.Context, cfg Config, errorHandler errorHandlerFunc, otlploggrpcNew otlploggrpcFactory) (OtelClient, error) {
```

Reply:
https://github.com/smartcontractkit/chainlink-common/commit/11c86c564cb226cc3b226c4e488b28aa63fa691a